src/Simulation/Native/build-native-simulator.ps1

```diff
@@ -10,25 +10,67 @@ if (-not (Test-Path $nativeBuild)) {
+    `
+    + "-fsanitize=address " `
+    + "-fsanitize=pointer-compare -fsanitize=pointer-subtract " `
+    + "-fsanitize-address-use-after-scope " `
+    + "-fno-omit-frame-pointer -fno-optimize-sibling-calls"
+
+    #+ "-fsanitize=unsigned-integer-overflow "      # TODO(rokuzmin, #884): Disable this option for _specific_ lines 
+                                                    # of code, but not for the whole binary.
+
 # There should be no space after -D CMAKE_BUILD_TYPE= but if we provide the environment variable inline, without
 # the space it doesn't seem to get substituted... With invalid -D CMAKE_BUILD_TYPE argument cmake silently produces
 # a DEBUG build.
 if (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
 {
-    Write-Host "On Windows build native simulator using the default C/C++ compiler (should be MSVC)"
-    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
+    $CMAKE_C_COMPILER = "-DCMAKE_C_COMPILER=clang.exe"
+    $CMAKE_CXX_COMPILER = "-DCMAKE_CXX_COMPILER=clang++.exe"
+
+    if ((!(Get-Command clang -ErrorAction SilentlyContinue) -and (choco find --idonly -l llvm) -contains "llvm") -or `
+        (Test-Path Env:/AGENT_OS)) {
+        # LLVM was installed by Chocolatey, so add the install location to the path.
+        $env:PATH = "$($env:SystemDrive)\Program Files\LLVM\bin;$env:Path"
+    }
+
+    cmake -G Ninja -D BUILD_SHARED_LIBS:BOOL="1" $CMAKE_C_COMPILER $CMAKE_CXX_COMPILER `
+        -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION"  ..
+        # Without `-G Ninja` we fail to switch from MSVC to Clang.
 }
 elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
 {
-    Write-Host "On Linux build native simulator using gcc (needed for OpenMP)"
-    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
+    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_C_COMPILER=clang-11 -D CMAKE_CXX_COMPILER=clang++-11 `
+        -D CMAKE_C_FLAGS_DEBUG="$SANITIZE_FLAGS" `
+        -D CMAKE_CXX_FLAGS_DEBUG="$SANITIZE_FLAGS" `
+        -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
 }
 elseif (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
 {
-    Write-Host "On MacOS build native simulator using gcc (needed for OpenMP)"
-    # `gcc`on Darwin seems to be a shim that redirects to AppleClang, to get real gcc, must point to the specific
-    # version of gcc we've installed.
-    cmake -D BUILD_SHARED_LIBS:BOOL="1" -D CMAKE_C_COMPILER=gcc-7 -D CMAKE_CXX_COMPILER=g++-7 -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ..
+    Write-Host "On MacOS build using the default C/C++ compiler (should be AppleClang)"
+
+    $OPENMP_PATH="/usr/local/opt/libomp"
+    $OPENMP_COMPILER_FLAGS="-Xpreprocessor -fopenmp -I$OPENMP_PATH/include -lomp -L$OPENMP_PATH/lib"
+    $OPENMP_LIB_NAME="omp"
```

Comment:
Not necessary for this change, but @idavis did some work on the QIR CMakeLists.txt to move these kinds of flags from the powershell script into the cmake, such that it becomes easier to compile without needing the powershell script (like from cargo or cmake directly). We don't need that immediately, but when we do want to enable that we should move these options into an `if (APPLE)` section in CMakeLists.txt.

Reply:
Transferred the OpneMP flags to CMake.